processing/src/main/java/io/druid/segment/QueryableIndex.java

```diff
@@ -31,17 +32,19 @@
  */
 public interface QueryableIndex extends ColumnSelector, Closeable
 {
-  public Interval getDataInterval();
-  public int getNumRows();
-  public Indexed<String> getAvailableDimensions();
-  public BitmapFactory getBitmapFactoryForDimensions();
-  public Metadata getMetadata();
-  public Map<String, DimensionHandler> getDimensionHandlers();
+  Interval getDataInterval();
+  int getNumRows();
+  Indexed<String> getAvailableDimensions();
+  BitmapFactory getBitmapFactoryForDimensions();
+  Metadata getMetadata();
+  Map<String, DimensionHandler> getDimensionHandlers();
+
+  SearchQueryDecisionHelper getDecisionHelper();
```

Comment:
Agree. How about making DecisionHelper based on index type in SearchStrategy? DecisionHelper can be used in only SearchStrategy as well as separating from index.